compiler/rustc_middle/src/ty/instance.rs

```diff
@@ -31,6 +31,13 @@ pub struct Instance<'tcx> {
     pub args: GenericArgsRef<'tcx>,
 }
 
+#[derive(Copy, Clone, PartialEq, Eq, Hash, Debug)]
```

Comment:
Document this please + the variants specifically. Also note that it's only currently used when KCFI is enabled.